### PR TITLE
chore: upgrade mui x-date-pickers to v6

### PR DESCRIPTION
**Pull request by maccuaa**
Continued the work from https://github.com/lookfirst/mui-rff/pull/1044 but focused on only upgrading x-date-pickers to v6 to limit the scope of the changes

### CODE REVIEW

**Comment by lookfirst** [comment]
This looks really good. Can you double check some of the snapshot changes? I see fullWidth and some other small changes that might be questionable (or could just be a result of the v6 changes, I don't know).

**Comment by maccuaa** [comment]
> This looks really good. Can you double check some of the snapshot changes? I see fullWidth and some other small changes that might be questionable (or could just be a result of the v6 changes, I don't know).

Good catch, i'll have a look

**Comment by maccuaa** [comment]
@lookfirst is the intention to force the textfield to be full width? Looks like that's what the old code was doing but is that correct? What if a user doesn't want a fullWidth text field? 

https://github.com/lookfirst/mui-rff/pull/1074/files#diff-76157577c12c983dee98352588bc51a2c2dc3906e3cd08994306c805b3df6fe0L58

**Comment by lookfirst** [comment]
I see what you're saying there. Agreed it should be removed.

File: test/__snapshots__/DatePicker.test.tsx.snap [hunk 1]
```diff
@@ -27,16 +27,18 @@ exports[`DatePicker renders without errors 1`] = `
...
           >
             <input
               aria-invalid="false"
+              autocomplete="off"
               class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
               id=":r0:"
+              inputmode="text"
               name="date"
-              placeholder="yyyy-mm-dd"
+              placeholder="YYYY-MM-DD"
```
Comment (lookfirst) [inline test/__snapshots__/DatePicker.test.tsx.snap:39]: Is this correct?

File: test/__snapshots__/DatePicker.test.tsx.snap [hunk 2]
```diff
@@ -27,16 +27,18 @@ exports[`DatePicker renders without errors 1`] = `
...
               required=""
-              type="tel"
+              type="text"
```
Comment (lookfirst) [inline test/__snapshots__/DatePicker.test.tsx.snap:41]: This looks correct, no idea how it got set to tel before!

File: test/__snapshots__/DatePicker.test.tsx.snap [hunk 3]
```diff
@@ -246,13 +250,13 @@ exports[`DatePicker turns red if empty and required 1`] = `
...
-          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
+          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
         >
           <label
-            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
-            data-shrink="false"
-            for=":r4:"
-            id=":r4:-label"
+            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
+            data-shrink="true"
```
Comment (lookfirst) [inline test/__snapshots__/DatePicker.test.tsx.snap:257]: I wonder if this is a result of the remove of fullWidth?

File: test/__snapshots__/DatePicker.test.tsx.snap (hunk 2 above)
Comment (maccuaa) [reply test/__snapshots__/DatePicker.test.tsx.snap:41]: was thinking the same thing! 😄

File: test/__snapshots__/DatePicker.test.tsx.snap (hunk 1 above)
Comment (maccuaa) [reply test/__snapshots__/DatePicker.test.tsx.snap:39]: i'll try to track down the cause of this change

**Comment by maccuaa** [comment]
> I see what you're saying there. Agreed it should be removed.

upon further review, this might be considered a "breaking" change since the other textfield-based components also force `fullWidth` to true.

https://github.com/lookfirst/mui-rff/blob/a7906524a0974da061cf4c7a6cd73e9e051edea8/src/Autocomplete.tsx#L183

https://github.com/lookfirst/mui-rff/blob/a7906524a0974da061cf4c7a6cd73e9e051edea8/src/TextField.tsx#L65

https://github.com/lookfirst/mui-rff/blob/a7906524a0974da061cf4c7a6cd73e9e051edea8/src/Select.tsx#L78

I'll put it back, don't want to change that behaviour in this PR

File: test/__snapshots__/DatePicker.test.tsx.snap (hunk 3 above)
Comment (maccuaa) [reply test/__snapshots__/DatePicker.test.tsx.snap:257]: looks like it was. That change went away when I added fullWidth back

**Comment by lookfirst** [comment]
great work. merged and released as 6.2.0. much appreciated!

**Comment by kjellski** [comment]
😞 sorry I never got to finish this PR - I wasn't able to continue after work pace picked up again...

Thanks to the both of you for finishing up :)

**Comment by maccuaa** [comment]
> 😞 sorry I never got to finish this PR - I wasn't able to continue after work pace picked up again...
> 
> Thanks to the both of you for finishing up :)

No worries! Thanks for getting it started, made it a lot easier to get these change in since you did most of the hard work already 😄

**Comment by scottdyoung** [comment]
> great work. merged and released as 6.2.0. much appreciated!

Apologies if I'm just blind, but looks like this change didn't make it into the 6.2.0 tag: https://github.com/lookfirst/mui-rff/blob/v6.2.0/package.json#L54

Great work on the project! Thanks for doing this

**Comment by lookfirst** [comment]
@scottdyoung That is so strange! It looks like you're correct. I have no idea what happened, but I'll release a new version asap.

**Pull request merged**